libs/hooks/src/mention/useMention.tsx

```diff
@@ -0,0 +1,170 @@
+        name: true,
+        nickname: true,
+        user_email: true,
+        user_tel: true,
+        user_icon: true,
+        user_memo: true,
+      },
+    });
+  },
+  ({ list }: UserAdminApi_SearchAllUsersInfoResponseBody) => list as UserInfo[]
+);
+
+interface LegacyMentionProps extends AMentionProps {
+  users: UserInfo[];
+  loading: boolean;
+  onSearch?: (text: string, prefix: string) => void;
+}
+
+const LegacyMention: React.FC<LegacyMentionProps> = ({
+  users,
+  loading,
+  onSearch,
+  ...restProps
+}) => {
+  return (
+    <Mentions onSearch={onSearch} loading={loading} {...restProps}>
+      {Array.isArray(users) &&
+        users.map((item) => (
+          <Mentions.Option value={item.name} key={item.name}>
+            <Avatar
+              src={item.user_icon}
+              size={24}
+              style={{
+                backgroundColor: item?.user_icon
+                  ? undefined
+                  : "rgb(0, 113, 235)",
+                marginRight: "10px",
+              }}
+            >
+              {!item.user_icon && item.name?.slice(0, 2)}
+            </Avatar>
+            {item.name}
+          </Mentions.Option>
+        ))}
+    </Mentions>
+  );
+};
+
+type UseMentionConfig = AMentionProps & {
+  debounceTime?: number;
+};
+
+type UseMentionReturnType = {
+  loading: boolean;
+  users: UserInfo[];
+  Mention: ReactElement<AMentionProps, typeof Mentions>;
+  updateConfig: (config: UseMentionConfig) => void;
+  updateUserOrInstanceId: (nameOrInstanceId: string) => void;
+};
+
+/**
+ *
+ * @param nameOrInstanceId
+ * @param config
+ * @return UseMentionReturnType
+ * @example
+ *
+ * function Todo () {
+ *   const {Mention,users,loading,updateConfig,updateUserOrInstanceId} = useMention("fakeUserName", config)
+ *   return <>
+ *     {Mention}
+ *   </>
+ * }
+ */
+export default function useMention(
+  nameOrInstanceId?: string,
+  config: UseMentionConfig = {}
+): UseMentionReturnType {
+  const { debounceTime = 300 } = config || {};
+  const [conf, setConf] = useState(config);
+  const [q, setQ] = useState(nameOrInstanceId);
+  const [users, setUsers] = useState<UserInfo[]>([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    if (!q) {
+      users?.length && setUsers([]);
+      return;
+    }
+    let isSubscribed = true;
+    setLoading(true);
+    getUserInfoByNameOrInstanceId(q)
+      .then(
+        (users) => {
+          if (isSubscribed) {
+            setUsers([...users]);
+          }
+        },
+        (err) => {
+          // eslint-disable-next-line no-console
+          console.error("Load user info failed:", err);
+        }
+      )
+      .finally(() => setLoading(false));
+    return () => {
+      isSubscribed = false;
+    };
+  }, [q]);
+
+  const handleMentionSearch = React.useRef(
+    debounce(async (text: string, prefix: string) => {
+      setQ(text);
+      config?.onSearch?.(text, prefix);
+    }, debounceTime)
+  ).current;
```

Comment:
使用 `useMemo`，这样才能响应相关参数的变化：

```suggestion
  const handleMentionSearch = React.useMemo(
    () =>
      debounce((text: string, prefix: string) => {
        setQ(text);
        config?.onSearch?.(text, prefix);
      }, debounceTime),
    [config?.onSearch, debounceTime]
  );
```